torch/quantization/fx/quantization_patterns.py

```diff
@@ -597,9 +618,12 @@ def convert(self,
                     if isinstance(self.conv, torch.nn.intrinsic._FusedModule):
                         float_conv = self.conv[0]
                     assert qconfig is not None
-                    float_conv.weight_post_process = qconfig.weight()
+                    weight_post_process = qconfig.weight()
                     # run weight observer
-                    float_conv.weight_post_process(float_conv.weight)
+                    weight_post_process(float_conv.weight)
+                float_conv._weight_qparams = get_qparam_dict(weight_post_process)
+                float_conv._register_state_dict_hook(_save_weight_qparams)
+                float_conv._register_load_state_dict_pre_hook(_load_weight_qparams, with_module=True)
```

Comment:
Have we considered having a `ReferenceConvNd` module instead, so these things would be defined on the class and easy to discover?  I could see a user printing out their model, seeing `nn.Conv2d` in the logs, not seeing any of these extra information in the class definition, and being really confused on where they came from.

Reply:
Agree that having different module would be more clear. I think not introducing new modules have other benefit though, for example simplifying the configurations: now we don't have quantized reference functions, what we need to specify is just operator and the type of operator, this is much simpler than having a mapping as we did before. Having a reference module here would make the configuration more complicated in this case.

This behavior will be clearly documented since it's needed for backend developers to lower the module correctly. If the concern is the logs I think maybe we could change the `_name` or something for the nn.Conv2d?

Reply:
the concern is that we will have `nn.Conv2d` with logic which is not present in the class definition, which some would consider technical debt.  Are there any ways around this which do not involve adding things to `nn.Conv2d` instances without adding them to the class definition?